src/storage/src/hummock/compactor/mod.rs

```diff
@@ -593,3 +623,160 @@ pub fn start_compactor(compactor_context: Arc<CompactorContext>) -> (JoinHandle<
+            _ = &mut shutdown_rx => {
+                tracing::info!("Compactor is shutting down");
+                return
+            }
+        };
+
+        match dispatch_task {
+            dispatch_compaction_task_request::Task::CompactTask(compact_task) => {
+                running_task_count.fetch_add(1, Ordering::SeqCst);
+                let (tx, rx) = tokio::sync::oneshot::channel();
+                let task_id = compact_task.task_id;
+                shutdown.lock().unwrap().insert(task_id, tx);
+                let mut output_object_ids: VecDeque<_> = VecDeque::new();
+                output_object_ids.extend(output_ids);
+                let sstable_object_id_manager =
+                    SharedComapctorObjectIdManager::new(output_object_ids);
+                let filter_key_extractor_manager =
+                    FilterKeyExtractorManagerFactory::ServerlessFilterKeyExtractorManager(
+                        FilterKeyExtractorBuilder::new(id_to_table),
+                    );
+                let (compact_task, table_stats) = shared_compact(
+                    compactor_metrics,
+                    compact_task,
+                    rx,
+                    sstable_store,
+                    parallel_compact_size_mb,
+                    filter_key_extractor_manager,
+                    worker_num,
+                    max_sub_compaction,
+                    memory_limiter,
+                    Box::new(sstable_object_id_manager.clone()),
+                    block_size_kb,
+                    object_store_recv_buffer_size,
+                    sstable_size_mb,
+                    task_progress_manager,
+                    bloom_false_positive,
+                    compactor_max_sst_size,
+                    compact_iter_recreate_timeout_ms,
+                    await_tree_reg,
+                )
+                .await;
+                shutdown.lock().unwrap().remove(&task_id);
+                running_task_count.fetch_sub(1, Ordering::SeqCst);
+                // todo: compactor pod send CompactTask via ReportCompactionTaskRequest
+                let report_compaction_task_request = ReportCompactionTaskRequest {
+                    event: Some(ReportCompactionTaskEvent::ReportTask(ReportSharedTask {
+                        compact_task: Some(compact_task),
+                        table_stats_change: to_prost_table_stats_map(table_stats),
+                    })),
+                };
+            }
+            dispatch_compaction_task_request::Task::VacuumTask(vacuum_task) => {
+                match Vacuum::handle_vacuum_task(
+                    sstable_store.clone(),
+                    &vacuum_task.sstable_object_ids,
+                )
+                .await
+                {
+                    Ok(_) => {
+                        // todo(wcy): report VacuumTask via ReportCompactionTaskRequest rpc.
+                        let report_compaction_task_request = ReportCompactionTaskRequest {
```

Comment:
WIP, will send `report_compaction_task_request ` to proxy via ReportCompactionTaskRequest.